lightwood/encoders/text/pretrained.py

```diff
@@ -0,0 +1,362 @@
+        self._frozen = frozen
+        self._batch_size = batch_size
+        self._epochs = epochs
+
+        # Model setup
+        self._tokenizer = custom_tokenizer
+        self._model = None
+        self.model_type = None
+
+        if model_name == "distilbert":
+            self._classifier_model_class = DistilBertForSequenceClassification
+            self._embeddings_model_class = DistilBertModel
+            self._tokenizer_class = DistilBertTokenizerFast
+            self._pretrained_model_name = "distilbert-base-uncased"
+
+        elif model_name == "albert":
+            self._classifier_model_class = AlbertForSequenceClassification
+            self._embeddings_model_class = AlbertModel
+            self._tokenizer_class = AlbertTokenizerFast
+            self._pretrained_model_name = "albert-base-v2"
+
+        elif model_name == "bart":
+            self._classifier_model_class = BartForSequenceClassification
+            self._embeddings_model_class = BartModel
+            self._tokenizer_class = BartTokenizerFast
+            self._pretrained_model_name = "facebook/bart-large"
+
+        else:
+            self._classifier_model_class = GPT2ForSequenceClassification
+            self._embeddings_model_class = GPT2Model
+            self._tokenizer_class = GPT2TokenizerFast
+            self._pretrained_model_name = "gpt2"
+
+        self.device, _ = get_devices()
+
+    def prepare(self, priming_data, training_data=None):
+        """
+        Prepare the encoder by training on the target.
+        """
+        if self._prepared:
+            raise Exception("Encoder is already prepared.")
+
+        # TODO: Make tokenizer custom with partial function; feed custom->model
+        if self._tokenizer is None:
+            # Set the default tokenizer
+            self._tokenizer = self._tokenizer_class.from_pretrained(
+                self._pretrained_model_name
+            )
+
+        # Replace empty strings with ''
+        priming_data = [x if x is not None else "" for x in priming_data]
+
+        # Check style of output
+
+        # Case 1: Categorical, 1 output
+        output_type = (
+            training_data is not None
+            and "targets" in training_data
+            and len(training_data["targets"]) == 1
+            and training_data["targets"][0]["output_type"]
+            == COLUMN_DATA_TYPES.CATEGORICAL
+        )
+
+        if self._custom_train and output_type:
+            print("Training model.")
+
+            # Prepare the priming data inputs with attention masks etc.
+            text = self._tokenizer(priming_data, truncation=True, padding=True)
+
+            # To train in the space, use labels as argmax.
+            labels = training_data["targets"][0]["encoded_output"].argmax(
+                dim=1
+            )  # Nbatch x N_classes
+            xinp = TextEmbed(text, labels)
+
+            # Pad the text tokens on the left (if padding allowed)
```

Comment:
Is this comment accurate?

Reply:
No - updated to reflect what this actually does; a comment like this should have been where the tokenizer call is. Will address.